helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/SoftConstraint.java

```diff
@@ -27,7 +27,7 @@
  * Evaluate a partition allocation proposal and return a score within the normalized range.
  * A higher score means the proposal is more preferred.
  */
-abstract class SoftConstraint {
+public abstract class SoftConstraint {
```

Comment:
Same as the Hard constraint, to keep it private, we can ensure the constraint concept is only known by the constraint-based algorithm. So I think we should keep it private.